File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
# Last updated: 8/9/2025, 2:26:49 AM
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        i = 0
        maxLen = 0
        for j in range(0, len(nums)):
            if nums[j] != 1:
                i = j + 1
            
            maxLen = max(maxLen, j - i + 1)
        return (maxLen)

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 3/22/2026, 10:34:08 PM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxNums = 0, curr = 0;
        for(int x: nums){
            if(x == 1){
                curr++;
            }else{
                curr = 0;
            }
            maxNums = max(curr, maxNums);
        }
        return maxNums;
    }
};